### PR TITLE
[Melodic] melodify

### LINKED ISSUE #181: cob_omni_drive_controller fails to compile with newer urdf versions
Compilation fails because newer urdf uses c++11 shared_ptr for urdf::JointConstSharedPtr

```
/var/tmp/portage/ros-kinetic/cob_omni_drive_controller-0.7.1/work/cob_omni_drive_controller-0.7.1/src/param_parser.cpp: In function ‘bool parseWheelGeom(WheelGeom&, XmlRpc::XmlRpcValue&, MergedXmlRpcStruct&, urdf::Model*)’:
/var/tmp/portage/ros-kinetic/cob_omni_drive_controller-0.7.1/work/cob_omni_drive_controller-0.7.1/src/param_parser.cpp:113:54: error: no match for ‘operator=’ (operand types are ‘boost::shared_ptr<const urdf::Joint>’ and ‘urdf::JointConstSharedPtr {aka std::shared_ptr<const urdf::Joint>}’)
         steer_joint = model->getJoint(geom.steer_name);
                                                      ^
In file included from /usr/include/boost/shared_ptr.hpp:17:0,
                 from /opt/ros/kinetic/include/ros/forwards.h:37,
                 from /opt/ros/kinetic/include/ros/common.h:37,
                 from /opt/ros/kinetic/include/ros/ros.h:43,
                 from /var/tmp/portage/ros-kinetic/cob_omni_drive_controller-0.7.1/work/cob_omni_drive_controller-0.7.1/include/cob_omni_drive_controller/UndercarriageCtrlGeomROS.h:21,
                 from /var/tmp/portage/ros-kinetic/cob_omni_drive_controller-0.7.1/work/cob_omni_drive_controller-0.7.1/src/param_parser.cpp:18:
```

@fmessmer FYI

### PR DESCRIPTION
**Pull request by jabbenseth**
replaces #203 #202 #201 to gain melodic compatiblity
fixes #181

### CODE REVIEW

**Review by fmessmer — CHANGES_REQUESTED** [review]
- what about #182 to fix #181?
- ~~`melodic` for `file` should be made mandatory in `.travis.yml`~~
- remove `allow_failure` from `.travis.yml`
- add CI and release info to README table

**Comment by jabbenseth** [comment]
> what about #182 to fix #181?

fixed in 3e70d29 and 93597fd

File: README.md [hunk 1]
```diff
@@ -3,21 +3,21 @@ cob_control
 
 ## ROS Distro Support
 
-|         | Indigo | Jade | Kinetic |
-|:-------:|:------:|:----:|:-------:|
-| Branch  | [`indigo_dev`](https://github.com/ipa320/cob_control/tree/indigo_dev) | [`indigo_dev`](https://github.com/ipa320/cob_control/tree/indigo_dev) | [`kinetic_dev`](https://github.com/ipa320/cob_control/tree/kinetic_dev) |
-| Status  |  supported | not supported |  supported |
-| Version | [version](http://repositories.ros.org/status_page/ros_indigo_default.html?q=cob_control) | [version](http://repositories.ros.org/status_page/ros_jade_default.html?q=cob_control) | [version](http://repositories.ros.org/status_page/ros_kinetic_default.html?q=cob_control) |
+|         | Indigo | Jade | Kinetic | Melodic |
```
Comment (fmessmer) [inline README.md:6]: `Lunar` column missing?

File: README.md [hunk 2]
```diff
@@ -3,21 +3,21 @@ cob_control
...
+|:-------:|:------:|:----:|:-------:|:-------:|
+| Branch  | [`indigo_dev`](https://github.com/ipa320/cob_control/tree/indigo_dev) | [`indigo_dev`](https://github.com/ipa320/cob_control/tree/indigo_dev) | [`kinetic_dev`](https://github.com/ipa320/cob_control/tree/kinetic_dev) | [`kinetic_dev`](https://github.com/ipa320/cob_control/tree/kinetic_dev) | [`kinetic_dev`](https://github.com/ipa320/cob_control/tree/kinetic_dev) |
```
Comment (fmessmer) [inline README.md]: or this is one column too many

File: README.md (hunk 1 above)
Comment (moriarty) [reply README.md:6]: Lunar is EOL. But so are Indigo and Jade.

**Comment by fmessmer** [comment]
@ipa-jba 
I took the liberty to rebase on top of #207 
I also fixed the table and tested on robot (kinetic, though) :ok: 

only thing, I would like to verify before merging is the links in the readme for the source and the debian jobs of the buildfarm, but that one is currently offline
:point_right: https://discourse.ros.org/t/jenkin-login-needed-for-buildfarm/9306

File: README.md (hunk 1 above)
Comment (fmessmer) [reply README.md:6]: it's just for the sake of completeness...

**Comment by jabbenseth** [comment]
> only thing, I would like to verify before merging is the links in the readme for the source and the debian jobs of the buildfarm, but that one is currently offline
>  https://discourse.ros.org/t/jenkin-login-needed-for-buildfarm/9306

seems ok now

**Review by fmessmer — APPROVED** [review]
final touches

**Pull request merged**